### PR TITLE
TINKERPOP-1628 Implement TraversalSelectStep

### PR DESCRIPTION
**Pull request by dkuppitz**
https://issues.apache.org/jira/browse/TINKERPOP-1628

This PR adds a `TraversalSelectStep` which supports the selection of dynamic / runtime-generated keys.

`docker/build.sh -t -i -n` passed.

VOTE: +1

### CODE REVIEW

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -74,6 +74,7 @@ This release also includes changes from <<release-3-2-8, 3.2.8>>.
 * TINKERPOP-1357 Centrality Recipes should mention pageRank and OLAP.
 * TINKERPOP-1489 Provide a Javascript Gremlin Language Variant
 * TINKERPOP-1586 SubgraphStrategy in OLAP
+* TINKERPOP-1628 Implement TraversalSelectStep.
```
Comment (spmallette) [inline CHANGELOG.asciidoc]: i know i pretty much always have false alarms about your changelog entries, but this one looks amiss. we generate this portion of the changelog on release so they shouldn't be edited. you just need an entry on 3.3.3 like we normally have.

File: docs/src/reference/the-traversal.asciidoc [hunk 2]
```diff
@@ -2338,6 +2338,22 @@ g.V(1).as("a").repeat(out().as("a")).times(2).select(last, "a")
...
+----
+g.withSideEffect("alias", ["marko":"okram"]).V().  <1>
+  values("name").sack(assign).                     <2>
+  optional(select("alias").select(sack()))         <3>
+----
+
+<1> Inject a name alias map and start the traversal from all vertices.
+<2> Select all `name` values and store them as the current traverser's sack value.
+<3> Optionally select the alias for the current name from the injected map.
+
```
Comment (spmallette) [inline docs/src/reference/the-traversal.asciidoc]: please add the two new `select()` methods to the list of "Additional References" below.

File: gremlin-test/features/map/Select.feature [hunk 3]
```diff
@@ -514,4 +514,20 @@ Feature: Step - select()
...
+      """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | d[3].l |
+      | d[1].l |
+      | d[3].l |
+      | d[3].l |
+      | d[1].l |
+      | d[1].l |
```
Comment (spmallette) [inline gremlin-test/features/map/Select.feature]: Could you please add one more test to include the other method you added - `select(Pop,Traversal)` ? I think it's worth exercising all these top level steps on the odd chance that something stops working somewhere.

**Comment by robertdale** [comment]
Finally got docs building again.
VOTE +1

**Pull request merged**